src/battlearena/SearchPlayers.js

```diff
@@ -0,0 +1,93 @@
+//@flow strict
+
+import * as React from 'react';
+
+type Props = {|
+	playerList: Array<string>,
+|};
+
+type State = {|
+	searchName: string,
+	searchList: Array<string>

```

Comment:
These should be identified by some Player object later on, because multiple players could have the same name (maybe, I guess possibly not, but we should be able to handle it if they do), and you need to identify who you are going to challenge in the challengePlayer function somehow.